### PR TITLE
add cinn bf16 support

### PR DESCRIPTION
**Pull request by lanxianghit**
### PR types

Others 
### PR changes

Others 
### Description

添加CINN与Paddle框架的BFloat16类型映射

### CODE REVIEW

**Review by Shixiaowei02 — DISMISSED** [review]

File: paddle/fluid/framework/paddle2cinn/transform_type.cc [hunk 1]
```diff
@@ -40,6 +40,7 @@ ::phi::DataType TransToPaddleDataType(const ::cinn::common::Type& type) {
   SET_TYPE_CASE_ITEM(UI32, UINT32)
   SET_TYPE_CASE_ITEM(UI64, UINT64)
 
+  SET_TYPE_CASE_ITEM(BF16, BFLOAT16)
```
Comment (thisjiang) [inline paddle/fluid/framework/paddle2cinn/transform_type.cc:43]: paddle/fluid/framework/paddle2cinn/transform_desc.cc
这文件也得改

File: setup.py [hunk 2]
```diff
@@ -1070,6 +1070,12 @@ def get_package_data_and_package_dir():
         if os.path.exists(cinn_fp16_file):
             shutil.copy(cinn_fp16_file, libs_path)
             package_data['paddle.libs'] += ['float16.h']
+        cinn_bf16_file = (
```
Comment (thisjiang) [inline setup.py:1073]: [setup.py.in#L627](https://github.com/PaddlePaddle/Paddle/blob/develop/python/setup.py.in#L627)
这儿应该也得改

File: paddle/fluid/framework/paddle2cinn/transform_type.cc (hunk 1 above)
Comment (lanxianghit) [reply paddle/fluid/framework/paddle2cinn/transform_type.cc:43]: 已修改

File: setup.py (hunk 2 above)
Comment (lanxianghit) [reply setup.py:1073]: 已修改

**Review by YuanRisheng — APPROVED** [review]

**Review by thisjiang — APPROVED** [review]
LGTM

**Review by zhangbo9674 — APPROVED** [review]
LGTM for  setup

**Pull request merged**